rosbag2_tests/test/rosbag2_tests/record_fixture.hpp

```diff
@@ -69,6 +69,20 @@ class RecordFixture : public TemporaryDirectoryFixture
     rclcpp::shutdown();
   }
 
+  void wait_for_metadata(int timeout = 5)
+  {
+    const auto metadata_path = rcpputils::fs::path{root_bag_path_} / "metadata.yaml";
+    const auto start_time = std::chrono::steady_clock::now();
+    while (std::chrono::steady_clock::now() - start_time < std::chrono::seconds(timeout)) {
```

Comment:
`... && rclcpp::ok()`

This way, this will actually unblock the thread when CTRL-C is triggered instead of forcing the process to be killed.